mera-weather-app/src/hooks/use-http.js

```diff
@@ -0,0 +1,36 @@
+import { useState, useCallback } from "react";
+
+const useHttp = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState(null);
+
+  const sendRequest = useCallback(async (requestConfig, applyData) => {
+    setIsLoading(true);
+    setError(null);
+    try {
+      const response = await fetch(requestConfig.url, {
+        method: requestConfig.method ? requestConfig.method : "GET",
+        headers: requestConfig.headers ? requestConfig.headers : {},
+        body: requestConfig.body ? JSON.stringify(requestConfig.body) : null,
+      });
+
+      if (!response.ok) {
+        throw new Error("Request failed!");
+      }
+
+      const data = await response.json();
+      applyData(data);
+    } catch (err) {
+      setError(err.message || "Something went wrong!");
+    }
+    setIsLoading(false);
```

Comment:
This should go inside of a `finally` block, to ensure that it's always called after `try` and `catch`.
It 99% won't happen here, but if an error was thrown inside of the `catch` block, line 26 would not execute as the code is written now.